SECURITY.md

```diff
@@ -0,0 +1,71 @@
+# Extensions Security Guidelines
+
+OpenSearch's support for extensions allows for taking already powerful use cases and expanding on them. With this increased functionality comes a larger surface area for misuse, vunerabilities, and malicious interactions.
+
+By capturing the current state of OpenSearch ecocsystem and the plans for extensions this document outlines several areas for enhancements, features, and practices to incorperate into extensions for OpenSearch.
+
+## Document Phases
```

Comment:
@dbwiddis Following your feedback added this section to capture how to approve/reject this document.